libs/sdk-ui-kit/src/@ui/@utils/keyboardNavigation.ts

```diff
@@ -19,56 +33,84 @@ export const makeMenuKeyboardNavigation = ({
     shouldPreventDefault = true,
     shouldStopPropagation = true,
 }: {
-    onFocusNext?: (event: React.KeyboardEvent) => void;
-    onFocusPrevious?: (event: React.KeyboardEvent) => void;
-    onFocusFirst?: (event: React.KeyboardEvent) => void;
-    onFocusLast?: (event: React.KeyboardEvent) => void;
-    onEnterLevel?: (event: React.KeyboardEvent) => void;
-    onLeaveLevel?: (event: React.KeyboardEvent) => void;
-    onSelect?: (event: React.KeyboardEvent) => void;
-    onClose?: (event: React.KeyboardEvent) => void;
-    onUnhandledKeyDown?: (event: React.KeyboardEvent) => void;
+    onFocusNext?: (event: T) => void;
+    onFocusPrevious?: (event: T) => void;
+    onFocusFirst?: (event: T) => void;
+    onFocusLast?: (event: T) => void;
+    onEnterLevel?: (event: T) => void;
+    onLeaveLevel?: (event: T) => void;
+    onSelect?: (event: T) => void;
+    onClose?: (event: T) => void;
+    onUnhandledKeyDown?: (event: T) => void;
 
     shouldPreventDefault?: boolean;
     shouldStopPropagation?: boolean;
 }) => {
-    function handleAction(event: React.KeyboardEvent, action?: (e: React.KeyboardEvent) => void) {
-        if (!action) {
-            return;
-        }
-
-        shouldPreventDefault && event.preventDefault();
-        shouldStopPropagation && event.stopPropagation();
-
-        action(event);
-    }
-
-    return (event: React.KeyboardEvent) => {
+    return (event: T) => {
         switch (event.code) {
             case "ArrowDown":
-                handleAction(event, onFocusNext);
+                handleAction(event, onFocusNext, shouldPreventDefault, shouldStopPropagation);
```

Comment:
This is super minor but you can change order of parameters and leave the `action` as the last one. You can than create the function wrapper above the `switch` as `const handleActionEvent = (action?: (e: T) => void) => handleAction(event, shouldPreventDefault, shouldStopPropagation, action);` and then in `switch` statement you can just call it as `handleActionEvent(onFocusPrevious/onLeaveLevel/...)` and will not need to repeat all those 3 parameters in every switch branch.

Feel free to ignore it if you don't see it the same way.